app/controllers/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class UsersController < ApplicationController
+
+  get '/users/:slug' do
+    @user = User.find_by_slug(params[:slug])
+    @tweets = Tweet.where(user_id: @user.id)
+    #binding.pry
+    erb :"/users/show"
+  end
+
+  get '/signup' do
+    # binding.pry
+    if logged_in?
+      redirect to '/tweets'
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+    #binding.pry
+    if params[:username] == "" || params[:password] == "" || params[:email] == ""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.